src/guide/essentials/watchers.md

```diff
@@ -245,41 +245,57 @@ export default {
 ```
 
 The initial execution of the handler function will happen just before the `created` hook. Vue will have already processed the `data`, `computed`, and `methods` options, so those properties will be available on the first invocation.
+  
+</div>
+
+<div class="composition-api">
+
+We can force a watcher's callback to be executed immediately by passing the `immediate: true` option:
+
+```js
+watch(source, (newValue, oldValue) => {
+  // ...
```

Comment:
We should add a bit of descriptive comment here, e.g. `executed immediately, then again when source changes...`